### PR TITLE
Extract one image from airgap bundle at a time

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: pkg/kotsadm/push_images.go [hunk 1]
```diff
@@ -332,14 +332,214 @@ func TagAndPushAppImages(imagesDir string, options types.PushImagesOptions) ([]k
...
+			defer os.Remove(tmpFile.Name())
+
+			_, err = io.Copy(tmpFile, tarReader)
+			if err != nil {
+				return errors.Wrapf(err, "failed to write file %q", header.Name)
+			}
+
+			tmpFile.Close()
+
+			if options.LogForUI {
```
Comment (salahalsaleh) [inline pkg/kotsadm/push_images.go:415]:
```suggestion
			if !options.LogForUI {
```

File: pkg/kotsadm/push_images.go [hunk 2]
```diff
@@ -332,14 +332,214 @@ func TagAndPushAppImages(imagesDir string, options types.PushImagesOptions) ([]k
...
+			}
+			defer tmpFile.Close()
+			defer os.Remove(tmpFile.Name())
+
+			_, err = io.Copy(tmpFile, tarReader)
+			if err != nil {
+				return errors.Wrapf(err, "failed to write file %q", header.Name)
+			}
+
+			tmpFile.Close()
```
Comment (salahalsaleh) [inline pkg/kotsadm/push_images.go:413]: there's already a `defer tmpFile.Close()` on line 405, not sure i get why this is added here

File: pkg/kotsadm/push_images.go [hunk 3]
```diff
@@ -332,14 +332,214 @@ func TagAndPushAppImages(imagesDir string, options types.PushImagesOptions) ([]k
...
+				err = image.CopyFromFileToRegistry(tmpFile.Name(), rewrittenImage.NewName, rewrittenImage.NewTag, rewrittenImage.Digest, registryAuth, reportWriter)
+				if err == nil {
+					break // image copy succeeded, exit the retry loop
+				} else {
+					options.Log.ChildActionWithoutSpinner("encountered error (#%d) copying image, waiting 10s before trying again: %s", i+1, err.Error())
+					time.Sleep(time.Second * 10)
+				}
+			}
+			if err != nil {
+				reportWriter.Write([]byte(fmt.Sprintf("+file.error:%s\n", err)))
```
Comment (salahalsaleh) [inline pkg/kotsadm/push_images.go:441]: does this need a `options.LogForUI` check as well?

File: pkg/kotsadm/push_images.go (hunk 2 above)
Comment (divolgin) [reply pkg/kotsadm/push_images.go:413]: this is to flush all data before passing file name to containers package

**Review by salahalsaleh — APPROVED** [review]

**Review by salahalsaleh — APPROVED** [review]

**Pull request merged**